integration_tests/src/main/python/data_gen.py

```diff
@@ -1044,6 +1045,19 @@ def gen_scalars_for_sql(data_gen, count, seed=None, force_no_nulls=False):
     spark_type = data_gen.data_type
     return (_convert_to_sql(spark_type, src.gen(force_no_nulls=force_no_nulls)) for i in range(0, count))
 
+class EmptyStringType(Enum):
+    ALL_NULL = 1
+    ALL_EMPTY = 2
+    MIXED = 3
+
+def mk_empty_str_gen(empty_type):
+    if (empty_type == EmptyStringType.ALL_NULL):
+        return NullGen(StringType())
+    elif (empty_type == EmptyStringType.ALL_EMPTY):
+        return StringGen("", nullable=False)
+    else:
+        return StringGen("", nullable=True)
```

Comment:
Nit: defensive programming
```suggestion
    elif (empty_type == EmptyStringType.MIXED):
        return StringGen("", nullable=True)
    else:
        raise AssertionError("unexpected empty type " + str(empty_type))
```
or alternatively make it a map lookup, e.g.:
```
empty_string_gens_map = {
  EmptyStringType.ALL_NULL : lambda: NullGen(StringType()),
  EmptyStringType.ALL_EMPTY : lambda: StringGen("", nullable=False)
  EmptyStringType.MIXED : lambda: StringGen("", nullable=True)
}

def mk_empty_str_gen(empty_type):
    return empty_string_gens_map[empty_type]()
```